lib/src/analyzers/lint_analyzer/rules/rules_list/ban_name/ban_name_rule.dart

```diff
@@ -0,0 +1,46 @@
+// ignore_for_file: public_member_api_docs
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../../../../../utils/node_utils.dart';
+import '../../../lint_utils.dart';
+import '../../../models/internal_resolved_unit_result.dart';
+import '../../../models/issue.dart';
+import '../../../models/severity.dart';
+import '../../models/common_rule.dart';
+import '../../rule_utils.dart';
+
+part 'visitor.dart';
+part 'utils/config_parser.dart';
+
+class BanNameRule extends CommonRule {
+  static const String ruleId = 'ban-name';
+
+  final List<_BanNameConfigEntry> _entries;
```

Comment:
@fzyzcjy please use Set for automatically remove duplicates

Reply:
Not automatically indeed. _BanNameConfigEntry not only has a name, but also a *description*. So sets will not prevent two entries with same name.

And if that happens, the plugin can still run (and it is indeed user's fault)